L1Trigger/Phase2L1ParticleFlow/plugins/L1SeedConePFJetProducer.cc

```diff
@@ -109,14 +139,38 @@ void L1SeedConePFJetProducer::produce(edm::StreamID /*unused*/,
                               }),
                work.end());
   }
-  std::sort(jets.begin(), jets.end(), [](l1t::PFJet i, l1t::PFJet j) { return (i.pt() > j.pt()); });
-  newPFJetCollection->swap(jets);
-  iEvent.put(std::move(newPFJetCollection));
+  return jets;
 }
 
-/////////////
-// DESTRUCTOR
-L1SeedConePFJetProducer::~L1SeedConePFJetProducer() {}
+std::vector<l1t::PFJet> L1SeedConePFJetProducer::processEvent_HW(std::vector<edm::Ptr<l1t::PFCandidate>>& work) const {
+  // The fixed point emulator
+  // Convert the EDM format to the hardware format, and call the standalone emulator
+  using namespace L1SCJetEmu;
+  Config config(_debug, _coneSize, _nJets);
+  std::vector<Particle> particles = convertEDMToHW(work);
+  std::vector<Jet> jets = emulateEvent(particles, config);
+  return convertHWToEDM(jets);
```

Comment:
In order to allow tracing back the PF cands, one thing you could do is:
 * in `convertEDMtoHW`, set also the [srcCand](https://github.com/thesps/cmssw/blob/L1PF_11_1_7_X_newfirmware_jetemulator/L1Trigger/Phase2L1ParticleFlow/src/newfirmware/dataformats/layer1_emulator.h#L108) to `edmParticle.get()` , and fill a `std::[unordered_]map<const l1t::PFCandidate *, edm::Ptr<l1t::PFCandidate>>` to remember the association
 *  in the emulator, return also the std::vector<Particle> of the constituents for each jet (either directly or perhaps making a l1ct::JetEmu class that extends l1ct::Jet class adding the vector of constituents?)
 * in `convertHWToEDM`, use the `srcCand` and map to get back the `edm::Ptr<l1t::PFCandidate>` and add the constituent to the jet.